### DISCUSSION TITLE
[Question] Protobuf deserialization for proto in json field

### DISCUSSION THREAD

**Discussion by ChristianGmw** [tchiotludo/akhq#708] [Q&A] [answered]
Hello,

just a quick question if what I want is possible.

The messages on the Kafka look like this, with the Proto object in the field "payload".

{"other_field1": 123, "other_field2": 456, "payload": "AkAAZwj1oNP6mC8SXlomCKq79BMQldblUhjrAyD......."}

Is this somehow possible with AKHQ to decode the proto in the json field?
I did not find anything in the doc that one could do such thing, but I thought wouldn't hurt to ask. 

thank you

**Comment by tchiotludo** [comment]
@xakassi any idea ? I'm a totally noob on protobuf :smile:

**Comment by xakassi** [comment, marked as answer]
Hi, @ChristianGmw !
For now it is not implemented. Protobuf deserialier class receives a topic's key or a topic's value as row bytes and decode them as one whole.
Actually I do not know how to implement logic you are looking for. It looks like that in your case we need to decode the whole bytes as a JSON and then additionally process some fields of the decoded JSON to decode them from Protobuf. But I'm not sure it will be correct.

Anyway for now it is not implemented.

**Comment by ChristianGmw** [comment]
Hi @xakassi ,
thanks for the answer.

It's not very important for us. We only really need to see the other Json fields and do not really need to see whats in the proto payload.
So we will just go with the current implementation.
